### ISSUE TITLE
Feature request: Command line options for reading a .nsd file and saving it as PNG/PDF/...

### ISSUE THREAD

**Issue by rpxrpxrpx** [fesch/Structorizer.Desktop#1171]
Hi.

It would be great if there were command line options to instruct Structorizer to read a .nsd file and export a picture or even code.

Example: `Structorizer --export-png myfile.nsd`
(Structorizer should read the .nsd file and export a .png file.)

It would be (among other scenarios) useful for the following: LaTeX documents that include structograms are stored in git repositories and only the .nsd files are checked in. With command line options as proposed above utilities like `make` could build all required .png files from the .nsd files automatically.

(This feature request is inspired by the `transfig` utility that transforms .fig files from xfig to lots of other formats.)

Thank you for reading this request.

**Comment by codemanyak** [comment]
@rpxrpxrpx 

> It would be great if there were command line options to instruct Structorizer to read a .nsd file and export a picture or even code.

Thank you for your proposal. The possibility to export code via command line parameters has been existing for many years. Have a look at the [user guide](view-source:https://help.structorizer.fisch.lu/index.php?menu=68&page=#batch_export). This even includes pseudo code and StrukTeX export, which both allow integration into LaTeX files. StrukTeX actually represents commands for graphical representation of the diagram.

It should be feasible to provide a command-line ("batch") conversion to picture files in e.g. png format, too.

**Comment by codemanyak** [comment]
Eventually, I found the time for some tests. It works for nsd files, and I am optimistic to get it working for arrangements (arr, arrz files), too. The proposed additional command line syntax will be:
`Structorizer (-i|--image) (png|emf|pdf|svg) [-s SETTINGSFILE] [-o OUTDIR] (NSDFILE|ARRSPEC|ARRZSPEC)...`

On this occasion, I bumped into an obsolete feature: The export as Flash file. I started a poll to abandon it.
